app/code/Magento/Catalog/Model/ResourceModel/Category/Collection.php

```diff
@@ -404,6 +460,23 @@ public function addRootLevelFilter()
         return $this;
     }
 
+    /**
+     * Add root category filter
+     *
+     * @return $this
+     */
+    public function addNavigationMaxDepthFilter()
```

Comment:
Adding a public method to a class or an interface is another thing which we need to get approval for from a Magento architect as it changes the public signature of the `@api`-annotated class.